module/VuFind/src/VuFind/Controller/Feature/UserContentTrait.php

```diff
@@ -0,0 +1,84 @@
+<?php
+
+/**
+ * User Content trait
+ *
+ * PHP version 8
+ *
+ * Copyright (C) The National Library of Finland 2025.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ *
+ * @category VuFind
+ * @package  Controller
+ * @author   Jaro Ravila <jaro.ravila@helsinki.fi>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org Main Site
+ */
+
+namespace VuFind\Controller\Feature;
+
+/**
+ * User Content trait
+ *
+ * @category VuFind
+ * @package  Controller
+ * @author   Jaro Ravila <jaro.ravila@helsinki.fi>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org Main Site
+ */
+trait UserContentTrait
+{
+    /**
+     * Get sort list for user content
+     *
+     * @param array  $options Array of sort options
+     * @param string $active  Currently active sort
+     *
+     * @return array
+     */
+    public function getSortList(array $options, string $active): array
+    {
+        $sortList = [];
+        foreach ($options as $key => $value) {
+            $sortList[$key] = [
+                'desc' => $value,
+                'url' => '?sort=' . urlencode($key),
+                'selected' => $active == $key,
+            ];
+        }
+        return $sortList;
+    }
+
+    /**
+     * Get record titles for user content
+     *
+     * @param Paginator $contents User content
+     *
+     * @return Paginator
+     */
+    public function getUserContentRecordTitles(\Laminas\Paginator\Paginator $contents)
+    {
+        $recordLoader = $this->serviceLocator->get(\VuFind\Record\Loader::class);
+        $ids = [];
+        foreach ($contents as $content) {
+            $ids[] = $content['source'] . '|' . $content['record_id'];
+        }
```

Comment:
Entirely optional, but I think you might be able to optimize this to something like the below, if I'm remembering my arrow function syntax correctly:

```suggestion
        $ids = array_map(fn ($content) => $content['source'] . '|' . $content['record_id'], $contents);
```

Reply:
Doesn't seem to work as `array_map` expects an array but `$contents` is a Paginator

Reply:
You could use `iterator_to_array` to make `$contents` into an array, and then it will probably work. Just a question of whether the functional approach is preferable or not -- I'll leave it up to you! :-)

Reply:
Switched to arrow function